### PR TITLE
don't push dup on --allow-dup

### PR DESCRIPTION
**Pull request by krzyzacy**
kops is checking things like kubelet SHA and aborts if the SHA changes, so let's try to skip the actual push part and see if it will WAI

/assign @BenTheElder @justinsb

### CODE REVIEW

File: lib/releaselib.sh [hunk 1]
```diff
@@ -563,8 +558,16 @@ release::gcs::push_release_artifacts() {
 
   # No need to check this for mock or stage runs
   # Overwriting is ok
+  # Return without error if --allow-up flag is set
   if ((FLAGS_nomock)) && ! ((FLAGS_stage)); then
-    release::gcs::destination_empty $dest || return 1
+    if release::gcs::destination_empty $dest; then
+      if ((FLAGS_allow_dup)); then
+        logecho "flag --allow-dup set, return without error"
```
Comment (BenTheElder) [inline lib/releaselib.sh]: bikeshed: s/return/exit

File: lib/releaselib.sh [hunk 2]
```diff
@@ -1066,8 +1069,16 @@ release::gcs::bazel_push_build() {
 
   # No need to check this for mock or stage runs
   # Overwriting is ok
+  # Return without error if --allow-up flag is set
   if ((FLAGS_nomock)) && ! ((FLAGS_stage)); then
-    release::gcs::destination_empty $gcs_destination || return 1
+    if release::gcs::destination_empty $dest; then
+      if ((FLAGS_allow_dup)); then
+        logecho "flag --allow-dup set, return without error"
```
Comment (BenTheElder) [inline lib/releaselib.sh]: same bikshed

**Comment by ixdy** [comment]
this is probably fine for fixing kops, but I think we've taken the wrong approach on this from the start - we shouldn't be continuously building kubernetes when nothing has changed.

File: lib/releaselib.sh (hunk 1 above)
Comment (krzyzacy) [reply lib/releaselib.sh]: it's not exiting the program though

Comment (BenTheElder) [reply lib/releaselib.sh]: shouldn't it? what else are we doing after not-uploading?

Comment (krzyzacy) [reply lib/releaselib.sh]: there are more stuff after https://github.com/kubernetes/release/blob/master/push-build.sh#L198

the docker images push and publish latest file pointer etc

**Comment by BenTheElder** [comment]
shouldn't we not do that if we're not uploading the build...?

**Comment by krzyzacy** [comment]
punt to https://github.com/kubernetes/test-infra/pull/6749

**Pull request closed without merging**